aeternity/config.py

```diff
@@ -0,0 +1,76 @@
+import os
+
+import requests
+
+
+class ConfigException(Exception):
+    pass
+
+class Config:
+    default_config = None
+
+    def __init__(self, local_port=None, internal_port=None, websocket_port=None, host='localhost'):
+        try:
+            if local_port is None:
+                local_port = os.environ.get('AE_LOCAL_PORT', 3013)
+            self.local_port = local_port
+            if internal_port is None:
+                internal_port = os.environ.get('AE_LOCAL_INTERNAL_PORT', 3113)
+            self.local_internal_port = internal_port
+            if websocket_port is None:
+                websocket_port = os.environ.get('AE_WEBSOCKET', 3114)
+            self.websocket_port = websocket_port
+        except KeyError:
+            raise ConfigException(
+                'You must either specify the Config manually, use '
+                'Config.set_default or provide the env vars AE_LOCAL_PORT, '
+                'AE_LOCAL_INTERNAL_PORT and AE_WEBSOCKET'
+            )
+
+        self.websocker_url = f'ws://{host}:{self.websocket_port}/websocket'
+        self.http_api_url = f'http://{host}:{self.local_port}/v2'
+        self.internal_api_url = f'http://{host}:{self.local_internal_port}/v2'
+
+        self.name_url = f'{self.http_api_url}/name'
+        self.pre_claim_url = self.internal_api_url + "/name-preclaim-tx"
+        self.claim_url = self.internal_api_url + "/name-claim-tx"
+        self.update_url = self.internal_api_url + "/name-update-tx"
+        self.transfer_url = self.internal_api_url + "/name-transfer-tx"
+        self.revoke_url = self.internal_api_url + "/name-revoke-tx"
```

Comment:
I would prefer handling the urls as properties as in `top_block_url`.

But in general, I think that the urls are not likely to be something that will have to be configurable anyway.

Reply:
yeah, this is mostly laziness on my part. `@property`s are probably nicer. In the end I didn't even use the URLs there, as they only appear once in the code. Still a code smell needing to be cleaned up.